src/utils/include/com/amazonaws/kinesis/video/utils/Include.h

```diff
@@ -1280,6 +1283,16 @@ typedef SEMAPHORE_HANDLE* PSEMAPHORE_HANDLE;
  */
 PUBLIC_API STATUS semaphoreCreate(UINT32, PSEMAPHORE_HANDLE);
 
+/**
+ * Create a semaphore object that starts with 0 count
+ *
+ * @param - UINT32 - IN - The permit count
+ * @param - PSEMAPHORE_HANDLE - OUT - Semaphore handle
+ *
+ * @return  - STATUS code of the execution
+ */
+PUBLIC_API STATUS semaphoreEmptyCreate(UINT32, PSEMAPHORE_HANDLE);
```

Comment:
If this was an initial commit of the library I would agree, but changing the constructor of such a core object now would deprecate a lot of code.

Reply:
The code will rot very fast if it's not agile to take changes like these. Higher level libraries have versioning for that purpose. Sure, it can cause a set of changes but all those changes are syntactical in nature. At the end of the day it's up to the owners of the codebase (which I am not) but I would recommend having a solid posture on how you keep the codebase agile even at the lowest layers.